bitcoin/src/blockdata/transaction.rs

```diff
@@ -713,28 +713,32 @@ impl Transaction {
     }
 
     /// Returns the size of this transaction excluding the witness data.
-    pub fn strippedsize(&self) -> usize {
-        let mut input_size = 0;
+    #[deprecated(since = "0.31.0", note = "Use Transaction::stripped_size() instead")]
+    pub fn strippedsize(&self) -> usize { Self::stripped_size(self).to_wu() as usize }
+
+    /// Returns the size of this transaction excluding the witness data.
+    pub fn stripped_size(&self) -> Weight {
+        let mut input_size: Weight = Weight::ZERO;
         for input in &self.input {
-            input_size += TxIn::BASE_WEIGHT.to_wu() as usize
-                + VarInt(input.script_sig.len() as u64).len()
-                + input.script_sig.len();
+            input_size += TxIn::BASE_WEIGHT
+                + Weight::from_wu_usize(VarInt(input.script_sig.len() as u64).len())
+                + Weight::from_wu_usize(input.script_sig.len());
         }
-        let mut output_size = 0;
+        let mut output_size = Weight::ZERO;
         for output in &self.output {
-            output_size += 8 + // value
```

Comment:
You've remove most of the other magic numbers, why not get rid of them all? Off the top of my head, while reviewing, I don't know what `8 // value` means.

Reply:
That sounds good.  Although I feel like there is enough going on for this PR.  When possible I'll do that in a separate PR.